zerver/lib/events.py

```diff
@@ -225,12 +225,12 @@ def fetch_initial_state_data(
             get_recursive_membership_groups(settings_user).values_list("id", flat=True)
         )
 
-    if want("realm_user_groups") or want("realm"):
-        realm_setting_group_ids = {
-            getattr(realm, setting_name + "_id")
-            for setting_name in Realm.REALM_PERMISSION_GROUP_SETTINGS
-        }
-
+    if (
+        want("realm_user_groups")
+        or want("realm")
+        or (want("stream") and include_streams)
+        or want("subscription")
+    ):
```

Comment:
This will fetch membership for all anonymous groups even if request is made for only streams data, but I don't think it would be trivial to first get all anonymous group IDs used for stream settings like it is for realm.